### ISSUE TITLE
Simplify the `load_textual_inversion` API

### ISSUE THREAD

**Issue by apolinario** [huggingface/diffusers#5733]
With Pivotal Tuning for SDXL it's common to train 2 tokens per text-encoder. The current API for loading it is as follows
```py
pipe.load_textual_inversion(state_dict["text_encoders_0"][0], token=["<s0>"], text_encoder=pipe.text_encoder, tokenizer=pipe.tokenizer)
pipe.load_textual_inversion(state_dict["text_encoders_0"][1], token=["<s1>"], text_encoder=pipe.text_encoder, tokenizer=pipe.tokenizer)
pipe.load_textual_inversion(state_dict["text_encoders_1"][0], token=["<s0>"], text_encoder=pipe.text_encoder_2, tokenizer=pipe.tokenizer_2)
pipe.load_textual_inversion(state_dict["text_encoders_1"][1], token=["<s1>"], text_encoder=pipe.text_encoder_2, tokenizer=pipe.tokenizer_2)
```
I think there could be better ways, maybe 

To me, the ideal scenario is: 
We standardize the name of the state dicts to match the name of the text encoders, so that gets automatically mapped out. If the state dict contains multiple inputs, we allow/require those multiple tokens

Suggested API
```py
pipe.load_textual_inversion("name/repo", weight_name="embeddings.safetensors", token=["<s0>", "<s1>"])
```
where in the `embeddings.safetensors` we have a state dict called `text_encoder` and another one called `text_encoder_2` and that gets mapped out accordingly to the respective text_encoder and tokenizer (if the user doesn't provide  `text_encoder` and `tokenizer` explicitly) 

I don't think it needs to be this particular API idea, but I think any move in the direction of simplifying the API would be great!

**Comment by patrickvonplaten** [comment]
Doesn't the following already work

```py
pipe.load_textual_inversion(state_dict["text_encoders_0"], token=["<s0>", "<s1>"], text_encoder=pipe.text_encoder, tokenizer=pipe.tokenizer)
pipe.load_textual_inversion(state_dict["text_encoders_1"], token=["<s0>", "<s1>"], text_encoder=pipe.text_encoder, tokenizer=pipe.tokenizer)
```

Also do we have lots of usage for pivotal tuning?

**Comment by apolinario** [comment]
> Doesn't the following already work

It doesn't. I tried that too before creating this issue, 
```bash
ValueError: You have passed a list of models of length 1, and list of tokens of length 2 Make sure both lists have the same length.
```

> Also do we have lots of usage for pivotal tuning?

Yes, as it has become one of the community most used ways to do fine-tuning via the sdxl-cog script. Currently it uses the custom class from sdxl-cog for loading them though -- however we plan on native supporting that with a community script for trianing, so having an easier API to load would be great

**Comment by patrickvonplaten** [comment]
> > Doesn't the following already work
> 
> It doesn't. I tried that too before creating this issue,
> 
> ```shell
> ValueError: You have passed a list of models of length 1, and list of tokens of length 2 Make sure both lists have the same length.
> ```
> 
> > Also do we have lots of usage for pivotal tuning?
> 
> Yes, as it has become one of the community most used ways to do fine-tuning via the sdxl-cog script. Currently it uses the custom class from sdxl-cog for loading them though -- however we plan on native supporting that with a community script for trianing, so having an easier API to load would be great

I see! We could make this work very easily! 

Ok if we simplify the loading logic then to:

```py
pipe.load_textual_inversion(state_dict["text_encoders_0"], token=["<s0>", "<s1>"], text_encoder=pipe.text_encoder, tokenizer=pipe.tokenizer)
pipe.load_textual_inversion(state_dict["text_encoders_1"], token=["<s0>", "<s1>"], text_encoder=pipe.text_encoder_2, tokenizer=pipe.tokenizer_2)
```

(and possibly to even more tokens). This means you'll never need more than two lines of code - one for each text encoder / tokenizer.

I really don't want to make it a one-liner because it means we need to create a class-specific abstraction for SDXL which bloats the code a lot.

**Comment by apolinario** [comment]
I’m ok with that! Two lines supporting n tokens one per text encoder sound like a good path forward, I’m ok with that so we don’t increase the complexity to make it a one liner

**Comment by patrickvonplaten** [comment]
@yiyixuxu are you interested in taking this issue? Essentially we should just make sure that we duplicate the `state_dict`, `text_encodre` and `tokenizer` to the same multiple as `token` if `n` tokens but only 1 state dict, text encoder and tokenizer are passed: https://github.com/huggingface/diffusers/blob/ecbe27a07f627b1d400f61e9d4bb9ca96521763c/src/diffusers/loaders/textual_inversion.py#L386

**Comment by patrickvonplaten** [comment]
@apolinario can you add an example checkpoint that should be better supported here?

**Comment by apolinario** [comment]
Yes, check this ones out:
- https://huggingface.co/LinoyTsaban/web_y2k_v3/tree/main
- https://huggingface.co/LinoyTsaban/3d_icon_LoRA/tree/main
- https://huggingface.co/LinoyTsaban/huggy_v28/tree/main

**Comment by patrickvonplaten** [comment]
cc @yiyixuxu

**Comment by apolinario** [comment]
Thanks @yiyixuxu ! Closed by #5837
